pkg/signer/signer.go

```diff
@@ -36,15 +36,40 @@ const TinkScheme = "tink"
 const MemoryScheme = "memory"
 const FileScheme = "file"
 
-func NewCryptoSigner(ctx context.Context, signer, kmsKey, tinkKmsKey, tinkKeysetPath, hcVaultToken, fileSignerPath, fileSignerPasswd string) (crypto.Signer, error) {
+func getHashFuncEllipticCurve(hashFuncStr string) (crypto.Hash, elliptic.Curve, error) {
+	switch hashFuncStr {
+	case "sha256":
+		return crypto.SHA256, elliptic.P256(), nil
+	case "sha384":
+		return crypto.SHA384, elliptic.P384(), nil
+	case "sha512":
+		return crypto.SHA512, elliptic.P521(), nil
+	default:
+		return 0, nil, fmt.Errorf("invalid hash algorithm - must be either sha256, sha384, or sha512")
+	}
+}
+
+func NewCryptoSigner(ctx context.Context, signer, kmsKey, tinkKmsKey, tinkKeysetPath, hcVaultToken, fileSignerPath, fileSignerPasswd, signerHashFunc string) (crypto.Signer, error) {
```

Comment:
I think this function is ready for a refactor but given that this PR already touches over ten files, it can be tackled in a follow up.